File: Bisq/src/main/java/com/joachimneumann/bisq/RawBisqNotificationViewModel.java
package com.joachimneumann.bisq;


import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;

import com.joachimneumann.bisq.Database.NotificationRepository;
import com.joachimneumann.bisq.Database.RawBisqNotification;

import java.util.List;

public class RawBisqNotificationViewModel extends AndroidViewModel {

    private NotificationRepository mRepository;
    public LiveData<List<RawBisqNotification>> rawBisqNotifications;

    public RawBisqNotificationViewModel (Application application) {
        super(application);
        mRepository = new NotificationRepository(application);
        rawBisqNotifications = mRepository.getAllRawBisqNotifications();
    }

    public void insert(RawBisqNotification rawBisqNotification) {
        mRepository.insert(rawBisqNotification);
    }

}

File: Bisq/src/main/java/com/joachimneumann/bisq/RawBisqNotificationViewModel.kt
package com.joachimneumann.bisq


import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData
import com.joachimneumann.bisq.Database.BisqNotification

import com.joachimneumann.bisq.Database.NotificationRepository

class RawBisqNotificationViewModel(application: Application) : AndroidViewModel(application) {

    private val mRepository: NotificationRepository
    var bisqNotifications: LiveData<List<BisqNotification>>

    init {
        mRepository = NotificationRepository(application)
        bisqNotifications = mRepository.allBisqNotifications
    }

    fun insert(bisqNotification: BisqNotification) {
        mRepository.insert(bisqNotification)
    }

}
